pioinstaller/__main__.py

```diff
@@ -88,8 +88,28 @@ def python():
         )
 
 
+@check.command("core")
+@click.option("--auto-upgrade/--no-auto-upgrade", is_flag=True, default=True)
+@click.option("--min-version", default=None)
```

Comment:
Please replace it by `--version-requirements`, where this value could be a version or version requirements. For example, `^4,!=4.2.1`, etc. There is Spec module in semver